reference-implementation/lib/abstract-ops/ecmascript.js

```diff
@@ -15,31 +16,17 @@ exports.CopyDataBlockBytes = (dest, destOffset, src, srcOffset, n) => {
   new Uint8Array(dest).set(new Uint8Array(src, srcOffset, n), destOffset);
 };
 
-// Not implemented correctly
 exports.TransferArrayBuffer = O => {
   assert(!exports.IsDetachedBuffer(O));
-  const transferredIshVersion = O.slice();
-
-  // This is specifically to fool tests that test "is transferred" by taking a non-zero-length
-  // ArrayBuffer and checking if its byteLength starts returning 0.
-  Object.defineProperty(O, 'byteLength', {
-    get() {
-      return 0;
-    }
-  });
-  O[isFakeDetached] = true;
-
-  return transferredIshVersion;
+  return ArrayBufferPrototypeTransferToFixedLength.call(O);
```

Comment:
I'm using `transferToFixedLength()` since we don't (yet) support [resizable `ArrayBuffer`s](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer#resizing_arraybuffers).

I think we can loosen that restriction, but we should do that in an *actual* spec change. 😉